astroquery/esa/utils/utils.py

```diff
@@ -7,11 +7,437 @@
+            url += "?TAPCLIENT=ASTROQUERY&format=votable_plain"
+        return super()._request(method, url, **kwargs)
+
+
+def get_coord_input(value, msg):
+    """
+    Auxiliary method to parse the coordinates
+
+    Parameters
+    ----------
+    value: str or SkyCoord, mandatory
+        coordinates to be parsed
+    msg: str, mandatory
+        Value to be shown in the error message
+
+    Returns
+    -------
+    The coordinates parsed
+    """
+    if not (isinstance(value, str) or isinstance(value,
+                                                 commons.CoordClasses)):
+        raise ValueError(f"{msg} must be either a string or astropy.coordinates")
+    if isinstance(value, str):
+        c = commons.parse_coordinates(value)
+        return c
+    else:
+        return value
+
+
+def get_degree_radius(radius):
+    """
+    Method to parse the radius and retrieve it in degrees
+
+    Parameters
+    ----------
+    radius: number or Quantity, mandatory
+        radius to be transformed to degrees
+
+    Returns
+    -------
+    The radius in degrees
+    """
+    if radius is not None:
+        if isinstance(radius, Quantity):
+            return radius.degree
+        elif isinstance(radius, float):
+            return radius
+        elif isinstance(radius, int):
+            return float(radius)
+    raise ValueError('Radius must be either a Quantity or float value')
+
+
+def download_table(astropy_table, output_file=None, output_format=None):
+    """
+    Auxiliary method to download an astropy table
+
+    Parameters
+    ----------
+    astropy_table: Table, mandatory
+        Input Astropy Table
+    output_file: str, optional
+        File where the table will be saved
+    output_format: str, optional
+        Format of the file to be exported
+    """
+    astropy_table.write(output_file, format=output_format, overwrite=False)
+
+
+def execute_servlet_request(url, tap, *, query_params=None):
+    """
+    Method to execute requests to the servlets on a server
+
+    Parameters
+    ----------
+    url: str, mandatory
+        Url of the servlet
+    tap: PyVO TAP, mandatory
+        TAP instance from where the session will be extracted
+    query_params: dict, optional
+        Parameters to be included in the request
+
+    Returns
+    -------
+    The request with the modified url
+    """
+
+    if 'TAPCLIENT' not in query_params:
+        query_params['TAPCLIENT'] = 'ASTROQUERY'
+
+    # Use the TAPService session to perform a custom GET request
+    response = tap._session.get(url=url, params=query_params)
+    if response.status_code == 200:
+        return response.json()
+    else:
+        response.raise_for_status()
+
+
+def execute_post_request(url, tap, *, query_params=None):
+    response = tap._session.post(url=url, params=query_params)
+    response.raise_for_status()
+
+    return response
+
+
+def plot_result(x, y, x_title, y_title, plot_title, *, error_x=None, error_y=None, log_scale=False):
```

Comment:
We don't usually do any add-on to data access functionality, such as plotting, so I'm a bit hesitant about these ones

Reply:
Totally agree, we should use astroquery to retrieve data only. I removed them.